qcodes/instrument_drivers/tektronix/Keithley_6500.py

```diff
@@ -0,0 +1,181 @@
+                           vals=Bool())
+
+        self.add_parameter('digits',
+                           get_cmd='DISP:VOLT:DC:DIG?', get_parser=int,
+                           set_cmd='DISP:VOLT:DC:DIG? {}',
+                           vals=Ints(min_value=4, max_value=7))
+
+        self.add_parameter('averaging_type',
+                           get_cmd=partial(self._get_mode_param,
+                                           'AVER:TCON', parse_output_string),
+                           set_cmd=partial(self._set_mode_param, 'AVER:TCON'),
+                           vals=Enum('moving', 'repeat'))
+
+        self.add_parameter('averaging_count',
+                           get_cmd=partial(self._get_mode_param,
+                                           'AVER:COUN', int),
+                           set_cmd=partial(self._set_mode_param, 'AVER:COUN'),
+                           vals=Ints(min_value=1, max_value=100))
+
+        self.add_parameter('averaging_enabled',
+                           get_cmd=partial(self._get_mode_param,
+                                           'AVER:STAT', parse_output_bool),
+                           set_cmd=partial(self._set_mode_param, 'AVER:STAT'),
+                           vals=Bool())
+
+        # Global parameters
+        self.add_parameter('display_enabled',
+                           get_parser=parse_output_bool,
+                           get_cmd='DISP:ENAB?',
+                           set_cmd='DISP:ENAB {}', set_parser=int, vals=Bool())
+
+        self.add_parameter('trigger_count',
+                           get_parser=int,
+                           get_cmd='ROUT:SCAN:COUN:SCAN?',
+                           set_cmd='ROUT:SCAN:COUN:SCAN {}',
+                           vals=MultiType(Ints(min_value=1, max_value=9999),
+                                          Enum('inf', 'default', 'minimum', 'maximum')))
+
+        self.add_parameter('trigger_delay',
+                           get_parser=float,
+                           get_cmd='TRIG:DEL?',
+                           set_cmd='TRIG:DEL {}',
+                           unit='s', vals=Numbers(min_value=0, max_value=999999.999))
+
+        self.add_parameter('trigger_source',
+                           get_cmd='TRIG:SOUR?',
+                           set_cmd='TRIG:SOUR {}',
+                           val_mapping={'immediate': 'NONE', 'timer': 'TIM', 'manual': 'MAN',
+                                        'bus': 'BUS', 'external': 'EXT'})
+
+        self.add_parameter('trigger_timer',
+                           get_parser=float,
+                           get_cmd='ROUT:SCAN:INT?',
+                           set_cmd='ROUT:SCAN:INT {}',
+                           unit='s', vals=Numbers(min_value=0.001, max_value=999999.999))
+
+        self.add_parameter('amplitude',
+                           get_cmd=self._read_next_value,
+                           unit='arb.unit')
+
+        self.add_function('reset', call_cmd='*RST')
```

Comment:
please don't use function, but a bound method instead. we are deprecating functions.